### PR TITLE
add Packages `py-vl-convert-python` `py-cairosvg` `py-cssselect2`; update `py-altair`

### PR DESCRIPTION
**Pull request by wu-kan**
add Packages `py-vl-convert-python` `py-cairosvg` `py-cssselect2`; update `py-altair`

### CODE REVIEW

File: var/spack/repos/builtin/packages/py-altair/package.py [hunk 1]
```diff
@@ -11,11 +11,16 @@ class PyAltair(PythonPackage):
 
     pypi = "altair/altair-4.2.0.tar.gz"
 
-    version("4.2.0", sha256="d87d9372e63b48cd96b2a6415f0cf9457f50162ab79dc7a31cd7e024dd840026")
+    version("5.1.2", sha256="7219708ec33c152e53145485040f428954ed15fd09b2a2d89e543e6d111dae7f",
+            url="https://files.pythonhosted.org/packages/17/16/b12fca347ff9d062e3c44ad9641d2ec50364570a059f3078ada3a5119d7a/altair-5.1.2-py3-none-any.whl", expand=False)
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-altair/package.py:15]: We prefer to use sdists (tar.gz) over bdists (whl)

File: var/spack/repos/builtin/packages/py-altair/package.py [hunk 2]
```diff
@@ -11,11 +11,16 @@ class PyAltair(PythonPackage):
...
+    version(
+        "4.2.0", sha256="d87d9372e63b48cd96b2a6415f0cf9457f50162ab79dc7a31cd7e024dd840026")
 
     depends_on("python@3.7:", type=("build", "run"))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-altair/package.py:19]: Deps need to be updated:
```toml
[build-system]
requires = ["hatchling"]
build-backend = "hatchling.build"

[project]
name = "altair"
authors = [{ name = "Vega-Altair Contributors" }]
dependencies = [
    "typing_extensions>=4.0.1; python_version<\"3.11\"",
    "jinja2",
    # If you update the minimum required jsonschema version, also update it in build.yml
    "jsonschema>=3.0",
    "numpy",
    # If you update the minimum required pandas version, also update it in build.yml
    "pandas>=0.25",
    "toolz",
    "packaging"
]
```

File: var/spack/repos/builtin/packages/py-cairosvg/package.py [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+
+class PyCairosvg(PythonPackage):
+    """Declarative statistical visualization library for Python"""
+
+    pypi = "CairoSVG/CairoSVG-2.7.1.tar.gz"
+
+    version("2.7.1", sha256="432531d72347291b9a9ebfb6777026b607563fd8719c46ee742db0aef7271ba0")
+
+    depends_on("python@3.5:", type=("build", "run"))
+    depends_on("py-setuptools@40.6:", type="build")
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-cairosvg/package.py:17]: Where does 40.6+ come from?

File: var/spack/repos/builtin/packages/py-cairosvg/package.py [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+    depends_on("py-entrypoints", type=("build", "run"))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-cairosvg/package.py:18]: Where does entrypoints come from?

File: var/spack/repos/builtin/packages/py-cairosvg/package.py [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+    depends_on("py-cairocffi", type=("build", "run"))
+    depends_on("py-cssselect2", type=("build", "run"))
+    depends_on("py-defusedxml", type=("build", "run"))
+    depends_on("py-pillow", type=("build", "run"))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-cairosvg/package.py]:
```suggestion
    depends_on("pil", type=("build", "run"))
```
We have a virtual dep for this

File: var/spack/repos/builtin/packages/py-cssselect2/package.py [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+from spack.package import *
+
+
+class PyCssselect2(PythonPackage):
+    """Declarative statistical visualization library for Python"""
+
+    pypi = "cssselect2/cssselect2-0.7.0.tar.gz"
+
+    version("0.7.0", sha256="fd23a65bfd444595913f02fc71f6b286c29261e354c41d722ca7a261a49b5969",
+            url="https://files.pythonhosted.org/packages/9d/3a/e39436efe51894243ff145a37c4f9a030839b97779ebcc4f13b3ba21c54e/cssselect2-0.7.0-py3-none-any.whl", expand=False)
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-cssselect2/package.py:15]: Should be using sdist, not bdist

Comment (adamjstewart) [reply var/spack/repos/builtin/packages/py-cssselect2/package.py:15]: Will need flit_core once you change this

File: var/spack/repos/builtin/packages/py-vl-convert-python/package.py [hunk 7]
```diff
@@ -0,0 +1,18 @@
...
+class PyVlConvertPython(PythonPackage):
+    """Declarative statistical visualization library for Python"""
+
+    pypi = "vl_convert_python/vl_convert_python-1.1.0.tar.gz"
+
+    version(
+        "1.1.0", sha256="cc2bf58e019a5b4796b2903af8eb952555a7701603e0752e9f9fe66627af5c2e")
+
+    depends_on("python@3.7:", type=("build", "run"))
+    depends_on("py-maturin", type=("build"))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-vl-convert-python/package.py]:
```suggestion
    depends_on("py-maturin@1.1:1", type="build")
```

**Review by adamjstewart — CHANGES_REQUESTED** [review]
These packages seem to be independent of each other. In that case, each package should be in a separate PR: https://spack.readthedocs.io/en/latest/contribution_guide.html

**Comment by wu-kan** [comment]
> These packages seem to be independent of each other. In that case, each package should be in a separate PR: https://spack.readthedocs.io/en/latest/contribution_guide.html

OK, I will create PR for each packagpe

**Pull request closed without merging**